src/config.c

```diff
@@ -862,33 +857,26 @@ void configGetCommand(client *c) {
     int matches = 0;
     int i;
 
-    for (standardConfig *config = configs; config->name != NULL; config++) {
-        int matched_conf = 0;
-        int matched_alias = 0;
-        for (i = 0; i < c->argc - 2 && (!matched_conf || !matched_alias); i++) {
+    dictIterator *di = dictGetIterator(configs);
+    dictEntry *de;
+    while ((de = dictNext(di)) != NULL) {
+        standardConfig *config = dictGetVal(de);
+        for (i = 0; i < c->argc - 2; i++) {
             robj *o = c->argv[2+i];
             char *pattern = o->ptr;
 
             /* Note that hidden configs require an exact match (not a pattern) */
-            if (!matched_conf &&
-                (((config->flags & HIDDEN_CONFIG) && !strcasecmp(pattern, config->name)) ||
-                 (!(config->flags & HIDDEN_CONFIG) && stringmatch(pattern, config->name, 1)))) {
-                addReplyBulkCString(c, config->name);
-                addReplyBulkSds(c, config->interface.get(config->data));
-                matches++;
-                matched_conf = 1;
-            }
-            if (!matched_alias && config->alias &&
```

Comment:
now `CONFIG GET *` will include both the alias and the real one?
maybe we wanna avoid it? i.e. for any specific argv argument, for each real config, somehow detect that both the alias and real config matched, return just one?
it may be a little bit complicated.

Reply:
The current behavior is to return both, I don't see a compelling reason to change that here.

Reply:
I think we must return both the name and the alias because otherwise a user looking for a specific config (via _grep_ for example) won't find it.
I also think it's best to avoid returning the same config twice if it matches multiple patterns. I made sure this won't happen in my multi get implementation. It didn't seem too complicated to achieve that.

Reply:
ok. so just to be clear, the code in this PR is ok from these perspectives.. right?